### PR TITLE
Update TestCase Asset tracking

### PR DESCRIPTION
**Pull request by tysmith**

### CODE REVIEW

File: grizzly/common/storage.py [hunk 1]
```diff
@@ -167,11 +168,21 @@ def clone(self):
...
             self.adapter_name,
-            self.input_fname,
-            self.time_limit,
-            self.timestamp,
+            input_fname=self.input_fname,
+            time_limit=self.time_limit,
+            timestamp=self.timestamp,
         )
-        result.assets = self.assets
+        result.assets = dict(self.assets)
```
Comment (jschwartzentruber) [inline grizzly/common/storage.py:175]: Up to you but I like `self.assets.copy()` instead for shallow clone.

File: grizzly/reduce/core.py [hunk 2]
```diff
@@ -559,7 +559,10 @@ def run(self, repeat=1, launch_attempts=3, min_results=1, post_launch_delay=0):
                                     # add target assets to test cases
                                     if not self.target.assets.is_empty():
                                         for test in reduction:
-                                            test.assets = self.target.assets
+                                            test.assets = dict(
+                                                self.target.assets.assets
```
Comment (jschwartzentruber) [inline grizzly/reduce/core.py:563]: DRY

File: grizzly/reduce/strategies/__init__.py [hunk 3]
```diff
@@ -157,12 +157,12 @@ def dump_testcases(self, testcases, recreate_tcroot=False):
             None
         """
         if recreate_tcroot:
-            rmtree(str(self._testcase_root))
+            rmtree(self._testcase_root)
             self._testcase_root.mkdir()
         for idx, testcase in enumerate(testcases):
             LOG.debug("Extracting testcase %d/%d", idx + 1, len(testcases))
-            testpath = self._testcase_root / f"{idx:0>3d}"
```
Comment (jschwartzentruber) [inline grizzly/reduce/strategies/__init__.py:164]: 👀

**Review by jschwartzentruber — APPROVED** [review]
Looks good, but `target.assets.assets` is not great. Maybe `target.asset_mgr.assets` ?

**Comment by tysmith** [comment]
> Looks good, but `target.assets.assets` is not great. Maybe `target.asset_mgr.assets` ?

Yeah I agree. I'll make that change in a follow up.

**Pull request merged**